zone/client_evolving_items.cpp

```diff
@@ -260,6 +260,25 @@ bool Client::DoEvolveCheckProgression(const EQ::ItemInstance &inst)
 
 	std::unique_ptr<EQ::ItemInstance> const new_inst(database.CreateItem(new_item_id));
 
+	if (!new_inst) {
+		return false;
+	}
+
+	if (RuleB(Character, EnableDiscoveredItems) && !IsDiscovered(new_inst->GetItem()->ID)) {
+		if (!GetGM()) {
+			DiscoverItem(new_inst->GetItem()->ID);
+		} else {
+			const std::string& item_link = database.CreateItemLink(new_inst->GetItem()->ID);
+			Message(
+				Chat::White,
+				fmt::format(
+					"Your GM flag prevents {} from being added to discovered items.",
+					item_link
+				).c_str()
+			);
+		}
```

Comment:
Can we add this code into a helper and call it once instead of duplicating these 10-15 or so lines

Reply:
Sure.